packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -553,6 +561,8 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
 
     protected abstract getInitialSnapshotDetails(): Promise<ISnapshotDetails>;
 
+    protected abstract getInitialGCDetails(): Promise<IGCDetails>;
+
```

Comment:
i would remove these altogether and just take protected read-only lazy promises in the constructor

Reply:
or just make then properties

---
In reply to: [546138404](https://github.com/microsoft/FluidFramework/pull/4484#discussion_r546138404) [](ancestors = 546138404)

Reply:
For remote context, there is a lot of work happening in this method. So I am a little hesitant in making this a property for now.